### PR TITLE
Improved error messages/handling for 2.18 migrations

### PR DESCRIPTION
**Pull request by allouis**
no-issue

We do not want to throw errors for edgecases, so we lot a message
telling the user to double check their settings.

### CODE REVIEW

File: core/server/data/migrations/versions/2.18/1-restore-settings-from-backup.js [hunk 1]
```diff
@@ -22,6 +22,11 @@ module.exports.up = (options) => {
     }, options);
 
     return fs.readdir(dataPath).then(function (files) {
+        return files
+    }).catch(function () {
```
Comment (kirrg001) [inline core/server/data/migrations/versions/2.18/1-restore-settings-from-backup.js]: why not global catch below?

Comment (allouis) [reply core/server/data/migrations/versions/2.18/1-restore-settings-from-backup.js]: Because it will catch errors with writing to the db

Comment (allouis) [reply core/server/data/migrations/versions/2.18/1-restore-settings-from-backup.js]: Do we want to catch this?

Comment (kirrg001) [reply core/server/data/migrations/versions/2.18/1-restore-settings-from-backup.js]: I'd not catch each error.
Just at the end and one general message.

Comment (kirrg001) [reply core/server/data/migrations/versions/2.18/1-restore-settings-from-backup.js]: no

Comment (kirrg001) [reply core/server/data/migrations/versions/2.18/1-restore-settings-from-backup.js]: If there is something wrong with his db in general, he will for sure notice when he starts his blog

Comment (kirrg001) [reply core/server/data/migrations/versions/2.18/1-restore-settings-from-backup.js]: Just my 2 cents. Can discuss in slack if you want :)

Comment (kirrg001) [reply core/server/data/migrations/versions/2.18/1-restore-settings-from-backup.js]: Let's keep like it is 👍

**Pull request merged**